src/cls/rbd/cls_rbd_types.h

```diff
@@ -73,13 +74,81 @@ struct MirrorImage {
   static void generate_test_instances(std::list<MirrorImage*> &o);
 
   bool operator==(const MirrorImage &rhs) const;
+  bool operator<(const MirrorImage &rhs) const;
 };
 
 std::ostream& operator<<(std::ostream& os, const MirrorImageState& mirror_state);
 std::ostream& operator<<(std::ostream& os, const MirrorImage& mirror_image);
 
 WRITE_CLASS_ENCODER(MirrorImage);
 
+enum MirrorImageStatusState {
+  MIRROR_IMAGE_STATUS_STATE_UNKNOWN         = 0,
+  MIRROR_IMAGE_STATUS_STATE_ERROR           = 1,
+  MIRROR_IMAGE_STATUS_STATE_SYNCING         = 2,
+  MIRROR_IMAGE_STATUS_STATE_STARTING_REPLAY = 3,
+  MIRROR_IMAGE_STATUS_STATE_REPLAYING       = 4,
+  MIRROR_IMAGE_STATUS_STATE_STOPPING_REPLAY = 5,
+  MIRROR_IMAGE_STATUS_STATE_STOPPED         = 6,
+};
+
+enum MirrorImageStatusFlag {
+  MIRROR_IMAGE_STATUS_FLAG_UP               = 1<<0,
+  MIRROR_IMAGE_STATUS_FLAG_DOWN             = 1<<1,
```

Comment:
Well, when the status is stored on disk, it is stored with no any flag set. And until you compare the status origin with current watchers you don't know if it is up or down. When a flag (UP or DOWN) is set you can be sure, that it has been checked against running watchers. If you remove DOWN flag, and the status comes with no UP, it would mean either it is DOWN or it has not been checked at all. Actually right now it is mostly useful for debugging and testing only, so I don't mind removing this. What do you prefer, a bool or flags with one flag (currently)?